### ISSUE TITLE
Service fabirc password decryption doesnt work if certificate is placed only in root and not in Personal store

### ISSUE THREAD

**Issue by LekhaSahdev** [Azure/service-fabric-issues#798]
We are using certificate based encryption for domain user password in Service manifest.xml. Even if we specify the store as Root, the service fails stating certificate & key is not available. If we install certificate in Personal store it works.
  <Principals>
    <Users>
      <User Name="ServiceAccountEncrypted" AccountType="DomainUser" AccountName="[Service_Account_Username]" Password="[Service_Account_Password]" PasswordEncrypted="true" />
    </Users>
  </Principals>
  <Policies>
    <SecurityAccessPolicies>
      <SecurityAccessPolicy ResourceRef="ConfigEncrypt" PrincipalRef="ServiceAccountEncrypted" GrantRights="Full" ResourceType="Certificate" />
    </SecurityAccessPolicies>
  </Policies>
  <Certificates>
    <SecretsCertificate X509StoreName="Root" X509FindValue="[ThumbPrint]" Name="ConfigEncrypt" />
  </Certificates>

When i go through the powershell command for decryption- Invoke-ServiceFabricDecryptText - It doesnt even have a parameter for Store name where as encrypt has store Invoke-ServiceFabricEncryptText - It has parameter for store name.

Please let us know why store name feature for decryption isnt working. If it is a defect with service fabric or if i am missing something

**Comment by RajeetN** [comment]
@LekhaSahdev, could you describe why you need certificate placed in root store? Is it actually a root cert?

**Comment by LekhaSahdev** [comment]
@RajeetN : Yes it is a trusted root certificate issued by our organisation. We are already using it for various other purposes. So we thought to use the same for Service Fabric

**Comment by RajeetN** [comment]
Service Fabric will require access to the private key of the certificate. Typically private key of root certificates is kept secure and not available on cluster nodes. Is the private key also available on all nodes? Then you need to copy the cert to localmachine/my store.

**Comment by LekhaSahdev** [comment]
@RajeetN : Got it. Thanks a lot. We just got confused for the reason that there was an option in appmanifest.xml to provide StoreName.

**Comment by induyadavusi** [comment]
I have placed my certificate in LocalMachine/My Store but still, I am facing the same issue that certificate and Private key not found.
